src/eval/array.rs

```diff
@@ -462,3 +468,9 @@ fn array_is_empty() -> EcoString {
 fn out_of_bounds(index: i64, len: i64) -> EcoString {
     eco_format!("array index out of bounds (index: {}, len: {})", index, len)
 }
+
+/// The out of bounds access error message when no default value was given.
+#[cold]
+fn no_default_and_out_of_bounds(index: i64, len: i64) -> EcoString {
+    eco_format!("no default value was specified and array index out of bounds (index: {}, len: {})", index, len)
```

Comment:
I'd prefer the actual error first and then the note about the default value (also for the other errors)
```suggestion
    eco_format!("array index out of bounds and no default value was specified (index: {}, len: {})", index, len)
```